src/App.js

```diff
@@ -1,22 +1,229 @@
+
+const http = axios.create()
 
 class App extends Component {
+  state = {
+    selectedLessons,
+    allLessons,
+    isOpen: false
+  }
+
+  handleStateChange = debounce((changes) => {
+    if (changes.hasOwnProperty('inputValue') && !isUndefined(changes['inputValue'])) {
+      return http
+        .get(`http://egghead.af:5000/api/v1/lessons?q=${changes['inputValue']}&size=20&page=1`)
+        .then(({data}) => data)
+        .then(allLessons => this.setState({allLessons}))
+    }
+  }, 250)
+
+  removeItem = (item) => {
+    this.setState({
+      selectedLessons: remove(this.state.selectedLessons, (itemToRemove) => !isEqual(itemToRemove, item))
+    })
+  }
+
+  selectItem = (item) => {
+    if(some(this.state.selectedLessons, item)) {
+      this.removeItem(item)
+    } else {
+      this.setState({
```

Comment:
This else handles adding an item to the list?

Reply:
just being lazy, it's more of a "toggle item selection" with this conditional in here.